java/ingest/ingest-runner/src/main/java/sleeper/ingest/impl/recordbatch/arrow/ArrowRecordBatchFactory.java

```diff
@@ -196,6 +184,18 @@ public ArrowRecordBatchFactory<Record> buildAcceptingRecords() {
         }
 
         public ArrowRecordBatchFactory<T> build() {
+            if (workingBufferAllocatorBytes < 1) {
+                throw new IllegalArgumentException("workingBufferAllocatorBytes must be positive");
+            }
+            if (maxBatchBufferAllocatorBytes < 1) {
+                throw new IllegalArgumentException("maxBatchBufferAllocatorBytes must be positive");
+            }
+            if (maxNoOfBytesToWriteLocally < 1) {
+                throw new IllegalArgumentException("maxNoOfBytesToWriteLocally must be positive");
+            }
+            if (maxNoOfRecordsToWriteToArrowFileAtOnce < 1) {
+                throw new IllegalArgumentException("maxNoOfRecordsToWriteToArrowFileAtOnce must be positive");
+            }
```

Comment:
It's pretty awkward having the validation split between the build method and the constructor like this. It should probably be all in one place. This was a problem in some of the PRs that have been merged as well, but I think we need to make it consistent.

Reply:
Thanks, i moved all the validation to the builder method.